components/director/internal/open_resource_discovery/validation.go

```diff
@@ -455,6 +522,39 @@ func validateJSONArrayOfObjects(arr interface{}, elementFieldRules map[string][]
 	return nil
 }
 
+func validateJSONObjects(obj interface{}, isOptional bool, elementFieldRules map[string][]validation.Rule, crossFieldRules ...func(gjson.Result) error) error {
+	jsonObj, ok := obj.(json.RawMessage)
+	if !ok {
+		return errors.New("should be json")
+	}
+
+	if len(jsonObj) == 0 && isOptional {
+		return nil
+	}
+
+	if !gjson.ValidBytes(jsonObj) && !isOptional {
```

Comment:
I don't quite get the logic behind the `isOptional`

For example: here if invalid JSON is passed but isOptional == true, then error won't be thrown and we continue with invalid json. Is this a desired behaviour?

I think we can reuse the validation part of `validateJSONArrayOfObjects` function